In [1]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error
from sklearn.metrics import r2_score
from sklearn.metrics import mean_absolute_error
from sklearn.metrics import mean_absolute_percentage_error

from sklearn.neural_network import MLPRegressor
from sklearn.datasets import make_regression
from sklearn.model_selection import train_test_split

In [2]:
df = pd.read_csv('wc-wo-outliers.csv')

In [3]:
y = df['results']
X = df.drop(columns=['results'])

X_train, X_test, y_train, y_test = train_test_split(
     X, y, test_size=0.20, random_state=1)

In [4]:
regr = MLPRegressor(random_state=1, hidden_layer_sizes=(58), 
                    max_iter=6000).fit(X_train, y_train)


In [5]:
regr.predict(X_test[:2])

array([12.6656768 , -1.59115221])

In [6]:
regr.score(X_test, y_test)

-0.43787398720360127

In [26]:
y_pred = regr.predict(X_test)

In [8]:
y_train_pred = regr.predict(X_train)

In [18]:
print('Neural Network Regression')
print('r2 score: '+str(r2_score(y_test, y_pred)))
print('RMSE : '+str(np.sqrt(mean_squared_error(y_test, y_pred))))
print('MSE: '+str(mean_squared_error(y_test, y_pred)))
print('MAE: '+str(mean_absolute_error(y_test, y_pred)))
print('MAPE: '+str(mean_absolute_percentage_error(y_test, y_pred)))

print('----------------')
print('r2 score: '+str(r2_score(y_train, y_train_pred)))
print('RMSE : '+str(np.sqrt(mean_squared_error(y_train, y_train_pred))))
print('MSE: '+str(mean_squared_error(y_train, y_train_pred)))
print('MAE: '+str(mean_absolute_error(y_train, y_train_pred)))
print('MAPE: '+str(mean_absolute_percentage_error(y_train, y_train_pred)))

Neural Network Regression
r2 score: -0.43787398720360127
RMSE : 10.277461899709312
MSE: 105.62622309997656
MAE: 8.754472829208435
MAPE: 1.2176183046988704
----------------
r2 score: 0.9991944877510535
RMSE : 0.25493237367886185
MSE: 0.06499051514953884
MAE: 0.1724085872621033
MAPE: 0.031775024245246475


## Neural Network Regression
- r2 score: -0.4642555623489475
- RMSE : 10.937460971051928
- MSE: 119.62805249328419
- MAE: 8.888064088667432
- MAPE: 1.1755273624373175

----------------
- r2 score: 0.9992320233044637
- RMSE : 0.25803390212941985
- MSE: 0.06658149464813501
- MAE: 0.17906714000929108
- MAPE: 0.030356789834672267


N## eural Network Regression
r2 score: -0.43787398720360127
RMSE : 10.277461899709312
MSE: 105.62622309997656
MAE: 8.754472829208435
MAPE: 1.2176183046988704
----------------
r2 score: 0.9991944877510535
RMSE : 0.25493237367886185
MSE: 0.06499051514953884
MAE: 0.1724085872621033
MAPE: 0.031775024245246475

In [19]:
data = {'y_test': y_test, 'y_pred': y_pred}
new_df = pd.DataFrame(data)
diff = (new_df['y_test'] - new_df['y_pred']).abs()
new_df['diff'] = diff
new_df = new_df.sort_values(by='diff', ascending=True)

In [20]:
new_df = new_df.sort_values(by='diff', ascending=True)

In [21]:
new_df['y_pred'] = round(new_df['y_pred'], 1)

In [23]:
new_df

y_test  y_pred       diff
29        4     3.4   0.602202
16       18    17.0   1.005290
95       23    24.7   1.713241
38       15    16.9   1.881709
190      24    26.4   2.416654
172       8     5.5   2.498264
163      14    10.7   3.311338
84       29    25.7   3.318794
119      28    31.5   3.526184
191      25    20.9   4.096669
31        2     6.2   4.203994
18       10     5.1   4.947129
174      22    16.9   5.097657
185      19    24.9   5.853953
116       7    13.3   6.338593
99        7    -0.3   7.325861
28        6    -1.6   7.591152
166       8    15.6   7.611181
56        8    16.1   8.122400
107      10    18.4   8.353419
120       3    11.8   8.781208
181      16    25.0   9.035015
53       17     7.7   9.266287
35        3    12.7   9.665677
40        7    17.0   9.951495
108       4    14.0   9.953150
44        6    -4.0   9.953425
169      18     7.7  10.343733
11        3    14.2  11.167608
4         5    16.6  11.554826
89       29    16.2  12.773952
105       4    16.8  12.824505
138       2    14.9  12.915743
113      16    30.0  14.019885
27        5    -9.6  14.562519
170      22     6.5  15.474934
127      18     2.1  15.911622
47       16    -1.2  17.214966
147      30     8.4  21.579099
102      17    -6.4  23.413582

In [24]:
new_df['diff'].mean()

8.754472829208435

In [25]:
new_df['diff'].median()

8.567313585848957